File: MissingNumber_day_12/missingNum.py
# Brute force method


def checkTheMissingNum(arr):

    n = len(arr)

    for i in range(n+1):
        if i not in arr:
            return i
    return -1

# Optimized method

def findMissingNumber(arr):

    n = len(arr)

    expectedNum = n * (n+1) // 2

    actualNum = 0 

    for num in arr:
        actualNum += 1

    return expectedNum - actualNum  

# set method

def find_missing_number(arr):
    seen = set(arr)

    for num in range(len(arr) + 1):
        if num not in seen:
            return num

    return -1

File: MissingNumber_day_12/missingNum.js
// Brute force method


function checkTheMissingNum(arr){

    const n = arr.length

    for(let i=0; i<=n; i++){
        if(!arr.includes(num)){
            return num
        }
    }
    return -1
}


// Optimized method

function findMissingNumber(arr){
        const n = arr.length

        const expectedSum = n *(n +1 ) / 2

        let actualNum = 0

        for(num of arr){
         actualNum += num
        }
        return expectedSum - actualNum
}